frontend/src/components/data-table/chart-transforms/chart-transforms.tsx

```diff
@@ -0,0 +1,681 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+
+import React, { useState, useMemo, useRef, useEffect } from "react";
+import { Button } from "@/components/ui/button";
+import {
+  ChartBarIcon,
+  InfoIcon,
+  SquareFunctionIcon,
+  TableIcon,
+  XIcon,
+} from "lucide-react";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { Tabs, TabsTrigger, TabsList, TabsContent } from "@/components/ui/tabs";
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectLabel,
+  SelectTrigger,
+  SelectValue,
+} from "../../ui/select";
+import type { z } from "zod";
+import { useForm, type UseFormReturn } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import {
+  ChartSchema,
+  DEFAULT_AGGREGATION,
+  DEFAULT_COLOR_SCHEME,
+} from "./chart-schemas";
+import { Form, FormControl, FormField, FormItem } from "@/components/ui/form";
+import { getDefaults } from "@/components/forms/form-utils";
+import { useAtom } from "jotai";
+import { type CellId, HTMLCellId } from "@/core/cells/ids";
+import { capitalize } from "lodash-es";
+import {
+  type TabName,
+  tabsStorageAtom,
+  ChartType,
+  tabNumberAtom,
+  CHART_TYPES,
+} from "./storage";
+import type { FieldTypesWithExternalType } from "../types";
+import { useAsyncData } from "@/hooks/useAsyncData";
+import { vegaLoadData } from "@/plugins/impl/vega/loader";
+import type { GetDataUrl } from "@/plugins/impl/DataTablePlugin";
+import { createVegaSpec } from "./chart-spec";
+import { useTheme } from "@/theme/useTheme";
+import { compile } from "vega-lite";
```

Comment:
can we lazy load this, or lazy load the whole `chart-transforms.tsx` files/component? this would ensure vega-lite (which is large, isn't in the first startup js payload)